full-service/src/db/transaction_log.rs

```diff
@@ -34,34 +30,34 @@ impl From<&TransactionLog> for TransactionId {
     }
 }
 
-impl From<&TxProposal> for TransactionId {
-    fn from(_tx_proposal: &TxProposal) -> Self {
-        Self::from(&_tx_proposal.tx.prefix)
-    }
-}
-
-impl From<&UnsignedTxProposal> for TransactionId {
-    fn from(_tx_proposal: &UnsignedTxProposal) -> Self {
-        Self::from(&_tx_proposal.unsigned_tx.tx_prefix)
+impl TryFrom<&TxProposal> for TransactionId {
+    type Error = &'static str;
+    fn try_from(_tx_proposal: &TxProposal) -> Result<Self, Self::Error> {
+        Self::try_from(_tx_proposal.payload_txos.clone())
     }
 }
 
-impl From<&Tx> for TransactionId {
-    fn from(src: &Tx) -> TransactionId {
-        Self::from(&src.prefix)
+impl TryFrom<&UnsignedTxProposal> for TransactionId {
+    type Error = &'static str;
+    fn try_from(_tx_proposal: &UnsignedTxProposal) -> Result<Self, Self::Error> {
+        Self::try_from(_tx_proposal.payload_txos.clone())
     }
 }
 
-impl From<&UnsignedTx> for TransactionId {
-    fn from(src: &UnsignedTx) -> TransactionId {
-        Self::from(&src.tx_prefix)
-    }
-}
-
-impl From<&TxPrefix> for TransactionId {
-    fn from(src: &TxPrefix) -> TransactionId {
-        let temp: [u8; 32] = src.digest32::<MerlinTranscript>(b"transaction_data");
-        Self(hex::encode(temp))
+impl TryFrom<Vec<OutputTxo>> for TransactionId {
+    type Error = &'static str;
+    fn try_from(_payload_txos: Vec<OutputTxo>) -> Result<Self, Self::Error> {
+        Ok(Self(
+            HexFmt(
+                _payload_txos
+                    .iter()
+                    .min_by_key(|txo| txo.tx_out.public_key)
```

Comment:
build_transaction and submit_transaction should share the same TransactionId but it is possible for the payload_txos (when there are more than one) to be present in a different order.  using min_by_key here (instead of e.g. picking the first element of the Vec) guarantees that even if the payload_txos within the TxProposal get sorted between calls to this impl, the TransactionId will remain the same.